project/Jdk9CompileDirectoriesPlugin.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+import java.io.File
+
+import sbt._
+import sbt.Keys._
+
+object Jdk9CompileDirectoriesPlugin extends AutoPlugin {
```

Comment:
This will make things work once we run on JDK9 - which is whenever the build is able to be sbt 1.0